tests/Integration/Behaviour/Features/Context/FeatureFlagFeatureContext.php

```diff
@@ -0,0 +1,159 @@
+    }
+
+    /**
+     * @When /^I (disable|enable) feature flag "(.+)"$/
+     */
+    public function enableFeatureFlag(string $state, string $name): void
+    {
+        $doctrineEntityManager = $this->getDoctrineEntityManager();
+
+        /** @var FeatureFlag $featureFlag */
+        $featureFlag = $doctrineEntityManager->getRepository('PrestaShopBundle:FeatureFlag')->findOneBy(['name' => $name]);
+
+        if ($state === 'enable') {
+            $featureFlag->enable();
+        } else {
+            $featureFlag->disable();
+        }
+
+        $doctrineEntityManager->flush();
+    }
+
+    /**
+     * @Then /^the feature flag "(.+)" state is (disabled|enabled)$/
+     */
+    public function assertFeatureFlagState(string $name, string $state): void
+    {
+        $doctrineEntityManager = $this->getDoctrineEntityManager();
+
+        /** @var FeatureFlag $featureFlag */
+        $featureFlag = $doctrineEntityManager->getRepository('PrestaShopBundle:FeatureFlag')->findOneBy(['name' => $name]);
+
+        if ($state === 'enabled' && !$featureFlag->getState()) {
+            throw new RuntimeException(sprintf('Feature flag %s is disabled although it was expected to be enabled', $name));
+        } elseif ($state === 'disabled' && $featureFlag->getState()) {
+            throw new RuntimeException(sprintf('Feature flag %s is enabled although it was expected to be disabled', $name));
+        }
+    }
+
+    /**
+     * @Then the system should reject my latest feature flag creation
+     */
+    public function featureFlagExcetion()
+    {
+        throw new PendingException();
+    }
+
+    /**
+     * @return EntityManager
+     */
+    protected function getDoctrineEntityManager(): EntityManager
+    {
+        /** @var EntityManager $doctrineEntityManager */
+        $doctrineEntityManager = CommonFeatureContext::getContainer()->get('doctrine.orm.entity_manager');
+
+        if (!$doctrineEntityManager->isOpen()) {
+            $doctrineEntityManager = CommonFeatureContext::getContainer()->get('doctrine')->resetManager();
+        }
+
+        return $doctrineEntityManager;
+    }
+
+    /**
+     * @AfterScenario
+     */
+    public function cleanFixtures()
+    {
+        $doctrineEntityManager = $this->getDoctrineEntityManager();
+
+        /** @var FeatureFlag $featureFlag */
+        $allFlags = $doctrineEntityManager->getRepository('PrestaShopBundle:FeatureFlag')->findAll();
+        foreach ($allFlags as $flag) {
+            $doctrineEntityManager->remove($flag);
+        }
+
+        $doctrineEntityManager->flush();
+    }
+
+    /**
+     * @AfterStep
+     */
+    public function clearEntityManager()
+    {
+        $this->getDoctrineEntityManager()->clear();
+    }
+
+    /**
+     * @Then /^I should be returned an error$/
+     */
+    public function assertGotErrorMessage()
+    {
+        if (!$this->latestResult instanceof \Exception) {
+            throw new \Exception('Latest action did not return an error');
```

Comment:
Move `\Exception` to `use` section

Reply:
Haha soon phpstan will tell this instead of you 😄

Reply:
But it doesn't so I do ;)

Reply:
Fixed by https://github.com/PrestaShop/PrestaShop/pull/23650/commits/bab4a0572083f021123539528de102d18074a36c